packages/@aws-cdk/cdk/lib/app.ts

```diff
@@ -2,6 +2,27 @@ import cxapi = require('@aws-cdk/cx-api');
 import { ConstructOrder, Root } from './core/construct';
 import { FileSystemStore, InMemoryStore, ISynthesisSession, SynthesisSession } from './synthesis';
 
+/**
+ * Custom construction properties for a CDK program
+ */
+export interface AppProps {
+  /**
+   * Automatically call run before the application exits
+   *
+   * If you set this, you don't have to call `run()` anymore.
+   *
+   * @default true
```

Comment:
It should be `@default "true" if CDK_OUTDIR is set`. Otherwise it should be false (for tests)

Reply:
Could do that, but it actually doesn't matter for tests.

All the queued up `run()`s get run, and they all do nothing because run() is idempotent now.

Reply:
Still feels like a safer approach, especially once we start copying asset files around during synthesis...